clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/SecurityResourceDefinition.java

```diff
@@ -0,0 +1,124 @@
+import org.jboss.as.clustering.controller.ResourceDescriptor;
+import org.jboss.as.clustering.controller.ResourceServiceHandler;
+import org.jboss.as.clustering.controller.SimpleResourceRegistration;
+import org.jboss.as.clustering.controller.SimpleResourceServiceHandler;
+import org.jboss.as.clustering.infinispan.subsystem.ComponentResourceDefinition;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.access.management.SensitiveTargetAccessConstraintDefinition;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.client.helpers.MeasurementUnit;
+import org.jboss.as.controller.operations.validation.StringLengthValidator;
+import org.jboss.as.controller.registry.AttributeAccess;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.jboss.dmr.ModelType;
+
+/**
+ * /subsystem=infinispan/remote-cache-container=X/component=security
+ *
+ * @author Radoslav Husar
+ */
+public class SecurityResourceDefinition extends ComponentResourceDefinition {
+
+    public static final PathElement PATH = pathElement("security");
+
+    public enum Attribute implements org.jboss.as.clustering.controller.Attribute {
+        SSL_CONTEXT,
+        ;
+
+        private final SimpleAttributeDefinition definition;
+
+        Attribute() {
+            SimpleAttributeDefinitionBuilder builder = createBuilder("ssl-context", ModelType.STRING);
+            this.definition = builder.setAllowExpression(false)
+                    .setCapabilityReference(new CapabilityReference(Capability.SECURITY_SSL, CommonUnaryRequirement.SSL_CONTEXT))
+                    .setValidator(new StringLengthValidator(1))
+                    .setAccessConstraints(SensitiveTargetAccessConstraintDefinition.SSL_REF)
+                    .build();
+        }
+
+        private static SimpleAttributeDefinitionBuilder createBuilder(String name, ModelType type) {
+            return new SimpleAttributeDefinitionBuilder(name, type)
+                    .setAllowExpression(true)
+                    .setRequired(false)
+                    .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+                    .setMeasurementUnit((type == ModelType.LONG) ? MeasurementUnit.MILLISECONDS : null)
+                    ;
+        }
+
+        @Override
+        public AttributeDefinition getDefinition() {
+            return this.definition;
+        }
+    }
+
+    enum Capability implements org.jboss.as.clustering.controller.Capability {
+        SECURITY_SSL("org.wildfly.clustering.infinispan.remote-cache-container.security.ssl-context", Void.class),
```

Comment:
This isn't a public capability, thus shouldn't have a service type.

Reply:
Thanks, fixed.